apps/web/src/app/components/Wallet/index.tsx

```diff
@@ -49,6 +51,15 @@ const Wallet = ({ close }: WalletProps) => {
     await xDisconnectAll();
   };
 
+  const handleSearchQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const query = e.target.value;
+    setSearchQuery(query);
+
+    const searchedChain = Object.values(xChainMap).find(chain => chain.name.toLowerCase() === query.toLowerCase());
```

Comment:
move the searching chain logic outside of the function. remove setSearchedXChainId

use the below code.
`searchedXChainId = Object.values(xChainMap).find(chain => chain.name.toLowerCase() === query.toLowerCase()).xChainId;`